### PR TITLE
183730528 - add pythnet to services

### PR DESCRIPTION
**Pull request by kamtarnowski**
#### What's this PR do?
PR updates services with pythnet.

#### How should this be manually tested?
Changes have impact on the following services:
SolanaRpcClient
ValidatorCheckActiveService
ValidatorSoftwareVersion

You can run them using the console in order to check if any fails. The database can be required to be populated with pythnet data.

#### What are the relevant tickets?
[- URL to the PivotalTracker ticket](https://www.pivotaltracker.com/story/show/183730528)

#### Task completed checklist:
- [] Is there appropriate test coverage?
- [] Did I take into consideration possible security vulnerabilities?
- [] Are the controllers secure?
- [] Was every important matter documented?

#### Things to watch out for
- [N] Does this PR have any migrations?
- [N] Does this PR add new dependencies?

### CODE REVIEW

File: app/services/validator_check_active_service.rb [hunk 1]
```diff
@@ -43,10 +43,13 @@ def should_be_destroyed?(validator)
...
-    if network == 'testnet'
-      @previous_epoch_testnet ||= @current_epoch.epoch - 1
-    else
+    case network
+    when "mainnet"
       @previous_epoch_mainnet ||= @current_epoch.epoch - 1
+    when "testnet"
+      @previous_epoch_testnet ||= @current_epoch.epoch - 1
+    when "pythnet"
+      @previous_epoch_pythnet ||= @current_epoch.epoch - 1
```
Comment (psuchta) [inline app/services/validator_check_active_service.rb:52]: I suggest changing the method to be more generic something like that
```
def previous_epoch(network)
    @previous_epochs ||= {}
    return @previous_epochs[network] if @previous_epochs[network].present?

    current_epoch = EpochWallClock.where(network: network).order(created_at: :desc).last
    @previous_epochs[network] = current_epoch - 1
  end
```

**Review by psuchta — CHANGES_REQUESTED** [review]

File: app/services/validator_check_active_service.rb [hunk 2]
```diff
@@ -43,10 +43,13 @@ def should_be_destroyed?(validator)
   # number of previous by network
   def previous_epoch(network)
     @current_epoch ||= EpochWallClock.where(network: network).order(created_at: :desc).last
```
[the remark this answers is not in the document]
Comment (kamtarnowski) [reply app/services/validator_check_active_service.rb:45]: Yeah it seems that.

File: app/services/solana_rpc_client.rb [hunk 3]
```diff
@@ -8,6 +8,7 @@ class SolanaRpcClient
   # default = SolanaRpcClient.new.default_client # uses cluster from initializers/solana_rpc_ruby
   # testnet = SolanaRpcClient.new.testnet_client # uses testnet client
   # mainnet = SolanaRpcClient.new.mainnet_client # uses mainnet client
+  # pythnet = SolanaRpcClient.new.pythnet_client # uses pythnet client
```
[the remark this answers is not in the document]
Comment (kamtarnowski) [reply app/services/solana_rpc_client.rb]: I did a rebase.

File: app/services/validator_check_active_service.rb (hunk 1 above)
Comment (kamtarnowski) [reply app/services/validator_check_active_service.rb]: Thanks, I did apply that.

**Review by psuchta — APPROVED** [review]

File: config/initializers/cluster.rb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 cluster_yml = YAML.load(File.read('config/cluster.yml'))
 MAINNET_CLUSTER_VERSION = cluster_yml['software_patch_mainnet']
 TESTNET_CLUSTER_VERSION = cluster_yml['software_patch_testnet']
+PYTHNET_CLUSTER_VERSION = cluster_yml['software_patch_pythnet']
```
Comment (psuchta) [inline config/initializers/cluster.rb:4]: We can also create CLUSTER_VERSION hash which will store clusters for each network

File: config/initializers/cluster.rb [hunk 5]
```diff
@@ -1,11 +1,15 @@
-cluster_yml = YAML.load(File.read('config/cluster.yml'))
-MAINNET_CLUSTER_VERSION = cluster_yml['software_patch_mainnet']
```
Comment (kbiala) [inline config/initializers/cluster.rb:2]: If you remove a constant, you need to update all places where this constant is being used.

**Review by kbiala — CHANGES_REQUESTED** [review]

File: config/initializers/cluster.rb (hunk 5 above)
Comment (kamtarnowski) [reply config/initializers/cluster.rb:2]: @kbiala  Where this constant is being used yet?

Comment (kbiala) [reply config/initializers/cluster.rb:2]: Run global search in the project. Both MAINNET_CLUSTER_VERSION and TESTNET_CLUSTER_VERSION are used in helpers.

Comment (kamtarnowski) [reply config/initializers/cluster.rb:2]: After pulling "pythnet-network" branch new occurrences of MAINNET_CLUSTER_VERSION and TESTNET_CLUSTER_VERSION were added.

**Pull request merged**